src/main/java/org/jenkinsci/plugins/workflow/cps/CpsGroovyShell.java

```diff
@@ -28,9 +35,51 @@
     /**
      * Use {@link CpsGroovyShellFactory} to instantiate it.
      */
+    @SuppressFBWarnings(value="DP_CREATE_CLASSLOADER_INSIDE_DO_PRIVILEGED", justification="irrelevant")
     CpsGroovyShell(ClassLoader parent, @CheckForNull CpsFlowExecution execution, CompilerConfiguration cc) {
         super(parent,new Binding(),cc);
         this.execution = execution;
+        try {
+            // Apparently there is no supported way to initialize a GroovyShell with a specified GroovyClassLoader.
```

Comment:
A bit confusing -- easier to follow if you say "GroovyShell creates its own GroovyClassLoader using the parent, no way to pass one in"